docs/mxnet_backend/performance_guide.md

```diff
@@ -0,0 +1,106 @@
+# Performance Tuning - Keras with MXNet Backend
+
+# Table of Contents
+1. [Objective](#objective)
+2. [Channels First Image Data Format for CNN](#channels-first-image-data-format-for-cnn)
+3. [Optimized MXNet binaries](#optimized-mxnet-binaries)
+4. [Using MXNet Profiler](#using-mxnet-profiler)
+5. [References](#references)
+
+# Objective
+In this document, we will discuss various tips and techniques to improve training/inference performance in Keras with 
+Apache MXNet backend.
+
+# Channels First Image Data Format for CNN
+
+When you install the `keras-mxnet`, by default, the following values are set.
+
+```json
+backend: mxnet
+image_data_format: channels_last
+```
+
+We strongly recommend changing the image_data_format to `channels_first`. MXNet is significantly faster on 'channels_first' data. 'channels_first' data format is optimal for training on NVIDIA GPUs with cuDNN, see [here](https://www.tensorflow.org/performance/performance_guide#data_formats) for more details.
+
+The default is set to 'channels_last' with an objective to be compatible with the majority of existing users of Keras-TensorFlow.
+
+You will see the following user warning with `channels_last` data format in MXNet backend.
+
+```
+UserWarning: MXNet Backend performs best with `channels_first` format. Using `channels_last` will significantly reduce performance due to the Transpose operations. For performance improvement, please use this API`keras.utils.to_channels_first(x_input)`to transform `channels_last` data to `channels_first` format and also please change the `image_data_format` in `keras.json` to `channels_first`.Note: `x_input` is a Numpy tensor or a list of Numpy tensor
+```
+
+Follow the below steps for moving to high performance 'channels_first' data format.
+
+## Step 1. Update Keras config file
+
+Update the `image_data_format` to `channels_first` in [Keras config file](https://keras.io/backend/#kerasjson-details), `keras.json`. You can find the Keras config file in your home directory `~/.keras/keras.json`.
+
+```json
+"image_data_format": "channels_first", 
+"backend": "mxnet"
+```
+
+## Step 2. Update data to channels first format
+
+If your data is in the channels last format (Ex: shape=(256, 256, 3)), you need to convert the data to channels first format (Ex: shape=(3, 256, 256)). 
+
+Keras-MXNet has a utility API [`keras.utils.to_channels_first()`](https://github.com/awslabs/keras-apache-mxnet/blob/master/keras/utils/np_utils.py#L55) that can take a single input or a batch of data in channels last format and return back the data in channels first format.
+
+
+```
+NOTE:
+    Keras dataset API is not consistent across the different dataset. It returns few datasets based on the 
```

Comment:
across different datasets. It returns a few datasets...

...and a few datasets in the channels...